src/fair/stats/__init__.py

```diff
@@ -566,3 +567,70 @@ def sample(self, n: int = 1) -> np.ndarray:
             samples.append(mBeta.sample())
 
         return np.vstack(samples)
+
+
+class GOF(mBeta):
+    """Goodness of fit test for mBeta distribution as described in section 2
+    https://arxiv.org/pdf/2003.06684
+    """
+
+    def __init__(self, null: mBeta, alt: mBeta) -> None:
+        """
+        Args:
+            null (mBeta): null mBeta distribution
+            alt (mBeta): alternative mBeta distribution
+        """
+        self.null = null
+        self.alt = alt
+
+    def _null_ecdf(self, n_samples: int = 100, t_samples: int = 100) -> ECDFResult:
+        """Empirical CDF of null distribution
+
+        Args:
+            n_samples (int, optional): Number of samples to draw. Defaults to 100.
+            t_samples (int, optional): Number of statistics to draw. Defaults to 100.
+
+        Returns:
+            ECDFResult: Empirical CDF of null distribution
+        """
+        samples = [
+            self._test_statistic(self.null, self.null, n_samples)
+            for i in range(t_samples)
+        ]
+
+        return stats.ecdf(samples)
+
+    def _test_statistic(
+        self, null: mBeta = None, actual: mBeta = None, n_samples: int = 100
+    ) -> float:
+        """Test goodness of fit
+
+        Args:
+            null (mBeta): null distribution
+            alt (mBeta): alternative distribution
+            n_samples (int, optional): Number of samples to draw. Defaults to 100.
+
+        Returns:
+            float: p-value
+        """
+        null = self.null if null is None else null
+        actual = self.alt if actual is None else actual
+        u_values = null.sample(n_samples)
+        v_values = actual.sample(n_samples)
```

Comment:
Why is null still called null but alt called actual?

Reply:
Alt used to be called actual, I didn't catch that one.